Buildings/Experimental/DHC/Loads/Steam/BaseClasses/ValveSelfActing.mo

```diff
@@ -0,0 +1,136 @@
+within Buildings.Experimental.DHC.Loads.Steam.BaseClasses;
+model ValveSelfActing "Ideal pressure reducing valve for steam heating systems"
+  extends Buildings.Fluid.Interfaces.PartialTwoPortInterface;
+  parameter Modelica.Units.SI.Pressure pb_nominal(
+    displayUnit="Pa",
+    min=101325)
+    "Nominal outlet pressure"
+    annotation (Dialog(group="Nominal condition"));
+
+  Buildings.Fluid.Movers.BaseClasses.IdealSource ideSou(
+    redeclare final package Medium = Medium,
+    final allowFlowReversal=allowFlowReversal,
```

Comment:
While setting `allowFlowReversal = false` within the extended `PartialTwoPortInterface` is not enforcing zero flow reversal, whereas there is an error-type assert warning in `IdealSource` that actually enforces zero flow reversal (and stops the simulation when `m_flow<0`). 

Hence, within `IdealSource`, should we set `final allowFlowReversal = allowFlowReversal`, which will trigger an error and stops the simulation, or should we set` final allowFlowReveral = true` only within this block so that the error is not triggered? 

Please let me know your thoughts.

Reply:
Well, I am a bit torn about this. So, in the current situation, if you were to propagate `allowFlowReversal = false` down to the ideal source component, the example fails when the flow reverse. If you set it to `true`, then when the source pressure drops below the sink pressure, the flow reverse. The valve is now a passthrough, with no pressure drop (port_b.p == port_a.p).

My issue with this is that it seems that in a real self-acting valve, this is not realistic. I believe (but please let me know if you think I'm wrong), that when the outlet pressure is greater than the inlet pressure, the valve closes, which result in no flow. If that's the case, then you should add a check valve (`Buildings.Fluid.FixedResistances.CheckValve`) somewhere on the path of the valve.

I have tried with the following parameters which seems to work:
```modelica
  Fluid.FixedResistances.CheckValve cheVal(
    redeclare final package Medium = Medium,
    final m_flow_nominal=m_flow_nominal,
    dpValve_nominal=1,
    rhoStd=Medium.density_pTX(
        pb_nominal,
        273.15 + 100,
        Medium.X_default),
    dpFixed_nominal=0)
```

In that case, the `allowFlowReversal` can be left to its default value (true).

The other proposition would be to actually set the `allowFlowReversal` to `false`, and actually let the model fail when the conditions are not handled by the model. In that case, you should just change the inlet parameters in the example so that it doesn't fail anymore.

What do you think?

Reply:
I think the second option suggested would be a better fit for a real self-acting steam pressure reducing valve (PRV).  

Steam systems will have a number of valves connected in series along the pipelines, and each valve has a specific function (e.g.: pressure reduction, back-flow prevention, mass flow control etc.). Moreover, having a check valve within the self-acting valve would still be a problem, as self-acting PRVs are typically added before the heat exchanger, and zero `m_flow` from PRV (due to check valve) will eventually fail the heat exchanger. Hence, setting the `allowFlowReversal` to `false` and letting the model fail would be the best option.  With this revision, I have removed the `assert ()` warning for negative `m_flow` as it is no longer needed. 

As suggested, I have changed inlet pressure signal in the example model to successfully run the simulations.